### DISCUSSION TITLE
Unable to render the puzzle pieces properly on mobile view

### DISCUSSION THREAD

**Discussion by alaksandarjesus** [flbulgarelli/headbreaker#74] [Q&A]
**Describe the bug**
I am trying to generate a mobile app puzzle game based on the jigsaw and the image was not splitted or rendered properly on solve.

I see duplicate on the left and right side

I also see the top of the image is rendered on the bottom.

**Screenshots**

![image](https://github.com/flbulgarelli/headbreaker/assets/4712552/57929aec-3212-405a-a18d-fd86f994e543)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version 120.0.6099.111

**settings code**

        import { Canvas, painters } from "headbreaker";

        const img = document.querySelector("img");

        var width =
        window.innerWidth ||
        document.documentElement.clientWidth ||
        document.body.clientWidth;

        var height =
        window.innerHeight ||
        document.documentElement.clientHeight ||
        document.body.clientHeight;
        let canvas;
        const dali = new Image();
        dali.onload = () => {
        canvas = new Canvas("puzzle", {
            width: width,
            height: height,
            image: dali,
            pieceSize: 100, proximity: 20,
            borderFill: 10, strokeWidth: 2,
            lineSoftness: 0.18,
            painter: new painters.Konva(),
        });
        canvas.autogenerate({
        });
        canvas.shuffle();
        canvas.draw();
        };

        dali.src = `https://picsum.photos/${width-100}/${height-100}.jpg`;
        console.log(dali.src)

        const btnSolve = document.querySelector("button.solve");
        btnSolve.addEventListener("click", function () {
        canvas.solve();
        console.log("button solve", canvas);
        });

**Comment by flbulgarelli** [comment]
Hi @alaksandarjesus! 👋

Have you tried adjusting the puzzle size to the width or height of the original image using `adjustImagesToPuzzleHeight` or `adjustImagesToPuzzleWidth`? You'll find several examples using that method in the demos, e.g. https://flbulgarelli.github.io/headbreaker/#autogenerated.

Hope that helps!

**Reply by alaksandarjesus** [reply]
I tried to follow the demo code but i did not get proper result. You can see the bottom trees are cut off.

I could also see the image was zoomed

        import { Canvas, painters } from "headbreaker";

        const img = document.querySelector("img");

        var width =
        window.innerWidth ||
        document.documentElement.clientWidth ||
        document.body.clientWidth;

        var height =
        window.innerHeight ||
        document.documentElement.clientHeight ||
        document.body.clientHeight;
        let canvas;
        const dali = new Image();
        dali.onload = () => {
        canvas = new Canvas("puzzle", {
            width: width,
            height: height,
            image: dali,
            pieceSize: 100, proximity: 20,
            borderFill: 10, strokeWidth: 2,
            lineSoftness: 0.18,
            painter: new painters.Konva(),
        });
        // canvas.adjustImagesToPuzzleHeight();
        canvas.adjustImagesToPuzzleWidth();
        canvas.autogenerate({
            horizontalPiecesCount: 6,
            verticalPiecesCount: 5
        });
        canvas.shuffle();
        canvas.draw();
        };

        dali.src = `https://picsum.photos/${width-100}/${height-100}.jpg`;
        console.log(dali.src)

        const btnSolve = document.querySelector("button.solve");
        btnSolve.addEventListener("click", function () {
        canvas.solve();
        console.log("button solve", canvas);
        });


![image](https://github.com/flbulgarelli/headbreaker/assets/4712552/2515a0c5-2a06-443f-8f9f-11c80d021f57)
